packages/slate-react/src/components/android/android-editable.tsx

```diff
@@ -0,0 +1,636 @@
+import React, {
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+import {
+  Descendant,
+  Editor,
+  Element,
+  Node,
+  NodeEntry,
+  Range,
+  Text,
+  Transforms,
+} from 'slate'
+import throttle from 'lodash/throttle'
+import scrollIntoView from 'scroll-into-view-if-needed'
+
+import {
+  IS_CHROME_LEGACY,
+  IS_EDGE_LEGACY,
+  IS_FIREFOX,
+  IS_SAFARI,
+} from '../../utils/environment'
+import { ReactEditor } from '../..'
+import { ReadOnlyContext } from '../../hooks/use-read-only'
+import { useSlate } from '../../hooks/use-slate'
+import { useIsomorphicLayoutEffect } from '../../hooks/use-isomorphic-layout-effect'
+import {
+  DOMElement,
+  DOMNode,
+  DOMRange,
+  DOMStaticRange,
+  getDefaultView,
+  isDOMNode,
+  isPlainTextOnlyPaste,
+} from '../../utils/dom'
+import {
+  EDITOR_TO_ELEMENT,
+  EDITOR_TO_RESTORE_DOM,
+  EDITOR_TO_WINDOW,
+  ELEMENT_TO_NODE,
+  IS_FOCUSED,
+  IS_READ_ONLY,
+  NODE_TO_ELEMENT,
+  PLACEHOLDER_SYMBOL,
+} from '../../utils/weak-maps'
+import { AndroidInputManager } from './android-input-manager'
+import { EditableProps } from '../editable'
+import { ErrorBoundary } from './ErrorBoundary'
+import useChildren from '../../hooks/use-children'
+
+// COMPAT: Firefox/Edge Legacy don't support the `beforeinput` event
+// Chrome Legacy doesn't support `beforeinput` correctly
+const HAS_BEFORE_INPUT_SUPPORT = !(
+  IS_FIREFOX ||
+  IS_EDGE_LEGACY ||
+  IS_CHROME_LEGACY
+)
+
+export const AndroidEditableNoError = (props: EditableProps): JSX.Element => {
+  return (
+    <ErrorBoundary>
+      <AndroidEditable {...props} />
+    </ErrorBoundary>
+  )
+}
+
+/**
+ * Editable.
+ */
+
+export const AndroidEditable = (props: EditableProps): JSX.Element => {
```

Comment:
Ah, I thought you just wanted to leave a note of a previous concern that had already been voiced and discussed.

The alternative design you are proposing was already considered when we discussed the need to integrate the android keyboard support into the core repository instead of the previous plugin design. In that discussion we got approved to embed the android keyboard support as long as we fork the `Editable` component. The approach you're proposing has merits but we will not be pursuing it.

While your concern is still relevant, it was already acknowledged and considered. The particular aspect of the design is not open for discussion.

Reply:
Hi @clauderic, I've opened up a separate issue to discuss a hook based Android implementation. I'm going to "resolve" these conversations, because we aren't going to be implementing your proposal in this PR, but it is not meant to mean that the issue of which approach is better is resolved. I've re-posted your summary in the issue here #4223 as well as adding some comments about why we won't be implementing that idea in this particular PR.